Fix NO_OPTIMIZE

Fix the following build failure when NO_OPTIMIZE is enabled (e.g. on
sparc) and -Ofast:

cc1: error: argument to '-O' should be a non-negative integer, 'g', 's' or 'fast'

Fixes:
 - http://autobuild.buildroot.org/results/e1a330e1a899fcdf4900e9156d62c90813321e30

Signed-off-by: Fabrice Fontaine <fontaine.fabrice@gmail.com>

diff -Nura snort-2.9.19.orig/configure.in snort-2.9.19/configure.in
--- snort-2.9.19.orig/configure.in	2022-08-09 20:21:00.236777320 +0200
+++ snort-2.9.19/configure.in	2022-08-09 20:29:02.260993315 +0200
@@ -1694,7 +1694,7 @@
 
 # Set to no optimization regardless of what user or autostuff set
 if test "x$NO_OPTIMIZE" = "xyes"; then
-    CFLAGS=`echo $CFLAGS | sed -e "s/-O./-O0/"`
+    CFLAGS=`echo $CFLAGS | sed -e "s/-O[0-9a-z]*/-O0/"`
 
     # in case user override doesn't include -O
     if echo $CFLAGS | grep -qve -O0 ; then
